python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -768,75 +814,74 @@ def run(
-        cat_ref[column_dummy] = np.zeros_like(ref.coord1)
-
-        # Add a boolean column for whether a match is classified correctly
-        # TODO: remove the assumption of a boolean column
-        extended_ref = cat_ref[config.column_ref_extended] == (not config.column_ref_extended_inverted)
-
-        extended_target = cat_target[config.column_target_extended].values >= config.extendedness_cut
-
-        # Define difference/chi columns and statistics thereof
-        suffixes = {MeasurementType.DIFF: 'diff', MeasurementType.CHI: 'chi'}
-        # Skip diff for fluxes - covered by mags
-        suffixes_flux = {MeasurementType.CHI: suffixes[MeasurementType.CHI]}
-        # Skip chi for magnitudes, which have strange errors
-        suffixes_mag = {MeasurementType.DIFF: suffixes[MeasurementType.DIFF]}
-        stats = {stat.name_short(): stat() for stat in (Median, SigmaIQR, SigmaMAD)}
-
-        for percentile in self.config.percentiles:
-            stat = Percentile(percentile=float(Decimal(percentile)))
-            stats[stat.name_short()] = stat
-
-        # Get dict of column names
-        columns, n_models = _get_columns(
-            bands_columns=config.columns_flux,
-            suffixes=suffixes,
-            suffixes_flux=suffixes_flux,
-            suffixes_mag=suffixes_mag,
-            stats=stats,
-            target=target,
-            column_dist=column_dist,
-        )
-
-        # Setup numpy table
-        n_bins = config.mag_num_bins
-        data = np.zeros((n_bins,), dtype=[(key, value) for key, value in columns.items()])
-        data['bin'] = np.arange(n_bins)
-
-        # Setup bins
-        bins_mag = np.linspace(start=config.mag_brightest_ref, stop=config.mag_faintest_ref,
-                               num=n_bins + 1)
-        data['mag_min'] = bins_mag[:-1]
-        data['mag_max'] = bins_mag[1:]
-        bins_mag = tuple((bins_mag[idx], bins_mag[idx + 1]) for idx in range(n_bins))
-
-        # Define temporary columns for intermediate storage
-        column_mag_temp = 'mag_temp'
-        column_color_temp = 'color_temp'
-        column_color_err_temp = 'colorErr_temp'
-        flux_err_frac_prev = [None]*n_models
-        mag_prev = [None]*n_models
-
-        columns_target = {
-            target.column_coord1: (
-                ref.column_coord1, target.column_coord1, coord1_target_err, False,
-            ),
-            target.column_coord2: (
-                ref.column_coord2, target.column_coord2, coord2_target_err, False,
-            ),
-            column_dist: (column_dummy, column_dist, column_dist_err, False),
-        }
-
-        # Cheat a little and do the first band last so that the color is
-        # based on the last band
+        data = None
         band_fluxes = [(band, config_flux) for (band, config_flux) in config.columns_flux.items()]
         n_bands = len(band_fluxes)
-        if n_bands > 0:
+        if self.config._compute_stats and (n_bands > 0):
```

Comment:
I imagine this whole `if` block needs an introductory `TODO` comment stating that it will be removed on DM-XXXXX as part of the deprecation process?